rotkehlchen/chain/evm/decoding/gearbox/gearbox_cache.py

```diff
@@ -0,0 +1,308 @@
+            method_name='getPoolsV3List',
+        )
+    except DeserializationError as e:
+        log.error(f'Gearbox data compressor returned an invalid pool address. {e}')
+        return None
+
+    gearbox_farming_pool_abi = evm_inquirer.contracts.abi('GEARBOX_FARMING_POOL')
+    new_pools = []
+    for pool_data in pools_data:
+        try:
+            pool_address = deserialize_evm_address(pool_data[0])
+        except DeserializationError:
+            log.error(f'Could not deserialize evm address {pool_data[0]} while decoding gearbox pool information')  # noqa: E501
+            continue
+
+        if pool_address in existing_pools:
+            continue
+        try:
+            pool_name: str = pool_data[4]
+            underlying = deserialize_evm_address(pool_data[1])
+            farming_pool_addresses: set[str] = set()
+            lp_token_addresses: set[str] = set()
+            zapper_tokens = {
+                de_addr
+                for data in pool_data[23]  # ZapperInfo[]
+                for addr in data[1:]  # ignore zapper address, only consider tokenIn and tokenOut
+                if (de_addr := deserialize_evm_address(addr)) not in {underlying, ETH_SPECIAL_ADDRESS}  # noqa: E501
+            }
+
+            calls = [
+                (
+                    (_contract := EvmContract(
+                        address=address,
+                        abi=gearbox_farming_pool_abi,
+                        deployed_block=0,
+                    )).address,
+                    _contract.encode(method_name='stakingToken'),
+                )
+                for address in zapper_tokens
+            ]
+            output = evm_inquirer.multicall_2(require_success=False, calls=calls)
+            for token_address, (is_farming_pool, _), in zip(zapper_tokens, output, strict=True):
+                if is_farming_pool:
+                    farming_pool_addresses.add(token_address)
+                else:
+                    lp_token_addresses.add(token_address)
+
+            new_pools.append(GearboxPoolData(
+                pool_address=pool_address,
+                pool_name=pool_name,
+                farming_pool_addresses=farming_pool_addresses,
+                lp_token_addresses=lp_token_addresses,
+                underlying_token=underlying,
+                chain_id=str(evm_inquirer.chain_id.value),
+            ))
+        except DeserializationError as e:
+            log.error(
+                f'Could not deserialize evm address while decoding gearbox pool {pool_address} '
+                f'information from data_compressor: {e}',
+            )
+
+    return new_pools
+
+
+def query_gearbox_data(
+        inquirer: 'EvmNodeInquirer',
+        cache_type: Literal[CacheType.GEARBOX_POOL_ADDRESS],
+) -> list[GearboxPoolData] | None:
+    """
+    Queries chain for all gearbox rewards pools and returns a list of the mappings not cached
+
+    May raise:
+    - RemoteError if failed to query chain
```

Comment:
where are the remoteErrors handled? Are they handled everywhere this logic is executed?

Reply:
its actually handled in the function itself, unnecessary comment, removed.